.dagger/sdk_java.go

```diff
@@ -15,8 +15,8 @@ const (
 	javaSDKVersionPomPath   = javaSDKPath + "/pom.xml"
 	javaSchemasDirPath      = javaSDKPath + "/dagger-codegen-maven-plugin/src/main/resources/schemas"
 	javaGeneratedSchemaPath = "target/generated-schema/schema.json"
-	javaVersion             = "17"
-	mavenVersion            = "3.9"
+	mavenImage              = "maven:3.9.9-eclipse-temurin-23-alpine"
```

Comment:
is there any reason to use java 23 rather than java 21 which is the latest LTS version?